lib/Convert.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2017-2017 Coburn Wightman
+// SPDX-License-Identifier: BSL-1.0
+
+#include <SoapySDR/Convert.hpp>
+
+SoapySDR::SourceFormatConverters Converters;
+
+bool SoapySDR::registerConverter(const std::string &sourceFormat, const std::string &targetFormat, SoapySDR::ConvertFunction converterFunction)
+{
+  if (Converters.count(sourceFormat) == 0)
+    {
+      Converters[sourceFormat][targetFormat] = converterFunction;
+    }
+  else
+    { 
+      if (Converters[sourceFormat].count(targetFormat) != 0)
+	{
+	  throw std::invalid_argument("conversion already registered: " + sourceFormat + " to " + targetFormat + ".");
+	}
+      else
+	{
+	  Converters[sourceFormat][targetFormat] = converterFunction;
+	}
+    }
+  
+  return true;
+}
+
+std::vector<std::string> SoapySDR::convertTargetFormats(const std::string &sourceFormat)
+{
+  std::vector<std::string> targets;
+
+  if (Converters.count(sourceFormat) == 0)
+    return targets;
+
+  for(SoapySDR::FormatConverters::iterator it = Converters[sourceFormat].begin() ; it != Converters[sourceFormat].end(); ++it)
+    {
+      std::string targetFormat = it->first;
+      targets.push_back(targetFormat);
+    }
+  
+  return targets;
+}
+
+std::vector<std::string> SoapySDR::convertSourceFormats(const std::string &targetFormat)
+{
+  std::vector<std::string> sources;
+
+  for(SoapySDR::SourceFormatConverters::iterator it = Converters.begin() ; it != Converters.end(); ++it)
```

Comment:
C++11 is your pal here. :-)

You can cut something down into a ranged for loop like so:

for (const auto &it : Converters)
{
  //it.first, it.second to access
}